Consistently stash the base URL in the configuration options

davehunt opened the pull request:
@bobsilverberg could you review this? It will mean that we only need to check `config.getoption` for the base URL even if it's specified in an INI file.

See also: https://github.com/mozilla/mcom-tests/blob/e0618d117000f338c37ebabd96de52469a4ca39b/tests/test_l10n_download_links.py#L13 and https://github.com/mozilla/Socorro-Tests/pull/341/files#r45722474

/cc @m8ttyB

bobsilverberg commented:
This looks good, @davehunt. r+

Would this be worth adding to the docs as a sort of "pro tip" - to let folks know where to look for the `base_url` if they need it?

davehunt commented:
In the process of adding the documentation I discovered this wasn't quite doing what we needed. I've now fixed it, and added a new test that verifies we can skip base on a base URL provided by a configuration file. @bobsilverberg r?

bobsilverberg commented:
Looks good, but Travis is failing.

davehunt commented:
I've seen a number of failures in Travis for BrowserStack connections timing out. I think we can safely ignore them, but I've now rebased this so we'll see if we get a pass this time.
